bin/fin

```diff
@@ -6287,6 +6344,631 @@ set_project_root_permissions ()
+	if_failed_error "Error importing DB"
+	echo-green "DB Pull Successful"
+}
+
+# Provider Pull Files Subcommand
+# Used specifically for pulling the db from a provider.
+provider_pull_files ()
+{
+	echo-yellow "Pulling Files..."
+
+	# Default directory for Drupal Files.
+	local drupal_files="sites/default/files/"
+	# Default directory for WordPress Files.
+	local wordpress_files="wp-content/uploads/"
+
+	# Default to Drupal Specifically.
+	local default_dir=${drupal_files}
+	# If find a wp-config.php file in the docroot set to wordpress_files
+	if [[ -f ${PROJECT_ROOT}/${DOCROOT}/wp-config.php ]]; then
+		default_dir=${wordpress_files}
+	fi
+
+	# Allow for this to be modified with the --FILES_DIR option.
+	FILES_DIR=${FILES_DIR:-$default_dir}
+	# Exclude specific items by default. Allow other rsync options to be included using --RSYNC_OPTIONS flag.
+	local rsync_options="${RSYNC_OPTIONS} --exclude='styles' --exclude='css' --exclude='js'"
+	provider_pull_find_function ${HOSTING_PLATFORM} 'files' 1
+	echo-green "File Pull Successful"
+}
+
+##############################
+### GENERIC PROVIDER FUNCTIONS
+##############################
+
+# Generic function for searching if function exists and executing.
+provider_pull_find_function ()
+{
+	# Providers name to attach to function.
+	local provider=$1
+	# Action for provider.
+	local action=$2
+	# Is this function required. Any value is acceptable.
+	local required=$3
+	# If required what is the error message if not found.
+	local required_message=${4:-"Pulling ${action} is currently not supported for ${provider}."}
+
+	# Check for help function for specific command
+	local pull_function="provider_pull_${provider}_${action}"
+	type "${pull_function}" >/dev/null 2>&1
+	if [ $? -eq 0 ]; then
+		# Execute function if found
+		${pull_function}
+	else
+		# If this was required and not found execute error message.
+		if [[ -n $required ]]; then
+			echo-error ${required_message}
+			exit 1
+		fi
+	fi
+}
+
+# Runs a git pull on the project.
+provider_pull_generic_git_pull ()
+{
+	local git_remote=${SYNC_GIT_REMOTE:-origin}
+	local current_branch=$(_exec "git rev-parse --abbrev-ref HEAD" | tr -d '\r')
+	local git_branch=${SYNC_GIT_BRANCH:-$current_branch}
+	echo-yellow "Pulling Code From ${git_remote} - ${git_branch}"
+	_exec "git pull -q ${git_remote} ${git_branch}"
+	if_failed_error "Error Pulling Code from ${git_remote} - ${git_branch}"
+}
+
+# Runs a generic drush rsync command.
+provider_pull_generic_drush_rsync ()
+{
+	echo-yellow "Downloading files from ${HOSTING_SITE} on the ${HOSTING_ENV} environment"
+	[[ ! -d ${PROJECT_ROOT}/${DOCROOT}/${FILES_DIR} ]] && mkdir -p ${PROJECT_ROOT}/${DOCROOT}/${FILES_DIR} > /dev/null
+	_exec "drush -y rsync \
+		@${HOSTING_SITE}.${HOSTING_ENV}:%files \
+		${PROJECT_ROOT}/${DOCROOT}/${FILES_DIR} \
+		--delete \
+		${rsync_options}"
+	if_failed_error "Error Downloading Files from ${HOSTING_SITE} on the ${HOSTING_ENV} environment"
```

Comment:
Downloading files might not be that big of an issue and maybe it makes sense to just `echo-error` here rather that fail and exit script.